I'm getting assertion error while comparing dicts. 
```
assert p.bisectors(2) == \
       {Point2D(0, 0): Ray2D(Point2D(0, 0), Point2D(0.71, 0.71)),
        Point2D(0, 3): Ray2D(Point2D(0, 3), Point2D(0.23, 2.0)),
        Point2D(1, 1): Ray2D(Point2D(1, 1), Point2D(0.19, 0.42)),
        Point2D(2, 0): Ray2D(Point2D(2, 0), Point2D(1.1, 0.38))}
```
> Traceback (most recent call last):
  File "/home/namannimmo/sympy/sympy/geometry/tests/test_polygon.py", line 415, in test_bisectors
    Point2D(2, 0): Ray2D(Point2D(2, 0), Point2D(1.1, 0.38))}
AssertionError

--- a/sympy/geometry/polygon.py
+++ b/sympy/geometry/polygon.py
@@ -1395,8 +1395,8 @@ class Polygon(GeometrySet):
             pts = list(reversed(pts))
         for v, a in p.angles.items():
             i = pts.index(v)
-            ray = Ray(pts[i], pts[i + 1]).rotate(a/2, v)
-            # normalize so distance from p1 to p2 is 1
+            p1, p2 = Point._normalize_dimension(pts[i], pts[i + 1])
+            ray = Ray(p1, p2).rotate(a/2, v)
             dir = ray.direction
             ray = Ray(ray.p1, ray.p1 + dir/dir.distance((0, 0)))
             if prec is not None:

--- a/sympy/geometry/tests/test_polygon.py
+++ b/sympy/geometry/tests/test_polygon.py
@@ -408,6 +408,7 @@ def test_bisectors():
     p1, p2, p3 = Point(0, 0), Point(1, 0), Point(0, 1)
     p = Polygon(Point(0, 0), Point(2, 0), Point(1, 1), Point(0, 3))
     q = Polygon(Point(1, 0), Point(2, 0), Point(3, 3), Point(-1, 5))
+    poly = Polygon(Point(3, 4), Point(0, 0), Point(8, 7), Point(-1, 1), Point(19, -19))
     t = Triangle(p1, p2, p3)
     assert t.bisectors()[p2] == Segment(Point(1, 0), Point(0, sqrt(2) - 1))
     assert p.bisectors()[Point2D(0, 3)] == Ray2D(Point2D(0, 3), \
@@ -416,6 +417,8 @@ def test_bisectors():
         Ray2D(Point2D(-1, 5), Point2D(-1 + sqrt(29)*(5*sin(acos(9*sqrt(145)/145)/2) + \
         2*cos(acos(9*sqrt(145)/145)/2))/29, sqrt(29)*(-5*cos(acos(9*sqrt(145)/145)/2) + \
         2*sin(acos(9*sqrt(145)/145)/2))/29 + 5))
+    assert poly.bisectors()[Point2D(-1, 1)] == Ray2D(Point2D(-1, 1), \
+        Point2D(-1 + sin(acos(sqrt(26)/26)/2 + pi/4), 1 - sin(-acos(sqrt(26)/26)/2 + pi/4)))
 
 def test_incenter():
     assert Triangle(Point(0, 0), Point(1, 0), Point(0, 1)).incenter \

--- a/sympy/geometry/tests/test_polygon.py
+++ b/sympy/geometry/tests/test_polygon.py
@@ -1,8 +1,9 @@
-from sympy import Abs, Rational, Float, S, Symbol, symbols, cos, pi, sqrt, oo
+from sympy import (Abs, Rational, Float, S, Symbol, symbols, cos, sin, pi, sqrt, \
+                    oo, acos)
 from sympy.functions.elementary.trigonometric import tan
 from sympy.geometry import (Circle, Ellipse, GeometryError, Point, Point2D, \
                             Polygon, Ray, RegularPolygon, Segment, Triangle, \
-                            are_similar,convex_hull, intersection, Line)
+                            are_similar,convex_hull, intersection, Line, Ray2D)
 from sympy.testing.pytest import raises, slow, warns
 from sympy.testing.randtest import verify_numerically
 from sympy.geometry.polygon import rad, deg
@@ -405,8 +406,16 @@ def test_reflect():
 
 def test_bisectors():
     p1, p2, p3 = Point(0, 0), Point(1, 0), Point(0, 1)
+    p = Polygon(Point(0, 0), Point(2, 0), Point(1, 1), Point(0, 3))
+    q = Polygon(Point(1, 0), Point(2, 0), Point(3, 3), Point(-1, 5))
     t = Triangle(p1, p2, p3)
     assert t.bisectors()[p2] == Segment(Point(1, 0), Point(0, sqrt(2) - 1))
+    assert p.bisectors()[Point2D(0, 3)] == Ray2D(Point2D(0, 3), \
+        Point2D(sin(acos(2*sqrt(5)/5)/2), 3 - cos(acos(2*sqrt(5)/5)/2)))
+    assert q.bisectors()[Point2D(-1, 5)] == \
+        Ray2D(Point2D(-1, 5), Point2D(-1 + sqrt(29)*(5*sin(acos(9*sqrt(145)/145)/2) + \
+        2*cos(acos(9*sqrt(145)/145)/2))/29, sqrt(29)*(-5*cos(acos(9*sqrt(145)/145)/2) + \
+        2*sin(acos(9*sqrt(145)/145)/2))/29 + 5))
 
 def test_incenter():
     assert Triangle(Point(0, 0), Point(1, 0), Point(0, 1)).incenter \

